CMakeLists.txt

```diff
@@ -15,13 +15,26 @@ set(NUM_CHUNKS 1)
 set(NUM_THREADS 160)
 set(NUM_VECTOR_LANES 1)
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
-        set(CMAKE_CXX_FLAGS "-fp-model=precise")
-elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
-        message("-- Detected Intel classic compiler which will be deprecated soon.")
-        message("-- It is recommended you use IntelLLVM compiler.")
-        set(CMAKE_CXX_FLAGS "-diag-disable=10441 -fp-model=precise")
-endif()
+if (SPECFEMPP_ENABLE_PYTHON)
+    message("-- Adding -fPIC flag for Python binding.")
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
+            set(CMAKE_CXX_FLAGS "-fp-model=precise -fPIC")
+    elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+            message("-- Detected Intel classic compiler which will be deprecated soon.")
+            message("-- It is recommended you use IntelLLVM compiler.")
+            set(CMAKE_CXX_FLAGS "-diag-disable=10441 -fp-model=precise -fPIC")
+    else()
+            set(CMAKE_CXX_FLAGS "-fPIC")
+    endif()
+else (SPECFEMPP_ENABLE_PYTHON)
```

Comment:
Is this syntax correct?

Reply:
I think so. I have tested with both pip and cmake.

Reply:
I actually like this syntax. It clears up what conditional you are in, and nothing would be put after else anyways.

I haven't seen it like that either though.